chapter05/red/exercise2.pl

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use List::Util qw( reduce );
+
+my $fn = shift @ARGV // 'coconet.dat';
+open my $fh, '<', $fn;
+
+my %stats;
+while( <$fh> ) {
+  next if /^\s*(#.*)?$/;
+  my ( $src, $dest, $bytes ) = split;
+  $stats{$src}{$dest} += $bytes;
+}
+
+close $fh;
+
+my @sources =
+  sort { $b->{size} <=> $a->{size} }
+  map { { addr => $_, size => reduce { $a + $b } values $stats{$_} } } keys %stats;
```

Comment:
You'll be able to do this too in a few chapters.

Also that comment reminds me of Persona games.